src/package_staleness.py

```diff
@@ -0,0 +1,127 @@
+import argparse
+import conda.cli.python_api
+import json
+from utils import (
+    get_dir_for_version,
+    get_semver,
+    get_installed_packages_and_conda_forge_sub_dir_from_env_out,
+    get_required_packages_from_env_in,
+)
+
+
+def _get_package_versions_in_conda_forge(installed_package_to_version_mapping,
+                                         package_to_conda_forge_subdirectories_mapping,
+                                         target_version) \
+        -> dict[str, str]:
+    package_to_version_mapping = {}
+    is_major_version_release = target_version.endswith('.0.0')
+    is_minor_version_release = target_version.endswith('.0') and not is_major_version_release
+    for package in installed_package_to_version_mapping:
+        # Execute a conda search api call in the linux-64 subdirectory
+        # packages such as pytorch-gpu are present only in linux-64 sub directory
+        version = installed_package_to_version_mapping[package]
+        versioning_split = version.split(".")
+        subdir_filter = '[subdir=' + package_to_conda_forge_subdirectories_mapping[package] + ']'
+        search_result = conda.cli.python_api.run_command('search', 'conda-forge::' + package +
+                                                         subdir_filter, '--json')
+        # Load the first result as json. The API sends a json string inside an array
+        package_metadata = json.loads(search_result[0])[package]
+        # Response is of the structure
+        # { 'package_name': [{'url':<someurl>, 'dependencies': <List of dependencies>, 'version':
+        # <version number>}, ..., {'url':<someurl>, 'dependencies': <List of dependencies>, 'version':
+        # <version number>}]
+        # We only care about the version number in the last index
+        package_version_in_conda = ''
+        if is_major_version_release:
+            package_version_in_conda = package_metadata[-1]['version']
+        elif is_minor_version_release:
+            package_major_version_prefix = versioning_split[0] + "."
+            package_version_in_conda = [x['version'] for x in package_metadata if
+                                        x['version'].startswith(package_major_version_prefix)][-1]
+        else:
+            package_minor_version_prefix = ".".join(versioning_split[:2]) + "."
+            package_version_in_conda = [x['version'] for x in package_metadata if
+                                        x['version'].startswith(package_minor_version_prefix)][-1]
+
+        package_to_version_mapping[package] = package_version_in_conda
+    return package_to_version_mapping
+
+
+def _generate_report(package_versions_in_conda_forge, package_versions_in_image, processor,
```

Comment:
`package_versions_in_conda_forge` -> `package_versions_upstream` or something, as we may use something other than Conda Forge as well in future.

Same for the header in the table that we create below.